omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java

```diff
@@ -171,14 +172,15 @@ private SimpleObject parseDisposition(DispositionDescriptor dispositionDescripto
 		return simpleObject;
 	}
 	
-	private SimpleObject parseDiagnosis(DiagnosisMetadata diagnosisMetadata, Obs obs) {
-		Diagnosis diagnosis = diagnosisMetadata.toDiagnosis(obs);
+	private SimpleObject parseDiagnosis(Encounter encounter) {
+		
+		Diagnosis diagnosis =Context.getDiagnosisService().getPrimaryDiagnosis(encounter);
```

Comment:
@ibacher   hi Ian, so i want to make sure that the returned diagnosis doesn't depend on the obs but rather the encounter,as shown in this line